Purpose: Make sure the utf-8 flag is not cleared from the tty.
Authour: Samuel Thibault
Fixes:   #547073
Status:  not submitted upstream yet

Index: trunk/src/init.c
===================================================================
--- trunk.orig/src/init.c	2009-10-25 15:29:44.000000000 +0100
+++ trunk/src/init.c	2009-10-25 15:30:30.000000000 +0100
@@ -721,7 +721,11 @@
 	/*
 	 *	Set pre and post processing
 	 */
-	tty.c_iflag = IGNPAR|ICRNL|IXON|IXANY;
+	tty.c_iflag = IGNPAR|ICRNL|IXON|IXANY
+#ifdef IUTF8 /* Not defined on FreeBSD */
+                      | (tty.c_iflag & IUTF8)
+#endif /* IUTF8 */
+            ;
 	tty.c_oflag = OPOST|ONLCR;
 	tty.c_lflag = ISIG|ICANON|ECHO|ECHOCTL|ECHOPRT|ECHOKE;
 
